addons/mail/static/src/core/common/record.js

```diff
@@ -1207,13 +1243,14 @@ export class Record {
     static _onChange(record, key, callback) {
         let proxy;
         function _observe() {
-            void proxy[key];
-            if (proxy[key] instanceof Object) {
-                void Object.keys(proxy[key]);
+            // access proxy[key] only once to avoid triggering reactive get() many times
+            const val = proxy[key];
+            if (val instanceof Object) {
```

Comment:
The `instanceof` operator checks whether the prototype of its second operand appears anywhere in the prototype chain of its first operand. But in JavaScript, there are "null-prototype objects," that is, objects that don't have a prototype, such as the ones you create by calling `Object.create(null)`. The current condition, which relies on prototypes, will fail to catch this kind of null-prototype object.
```javascript
console.log(Object.create(null) instanceof Object); // false
```
Maybe you have some reason to trust that the object here will never be a null-prototype object, but just to be on the safe side, I'd rather rewrite the check this way:
```suggestion
            if (typeof val === "object" && val !== null) {
```
(extra check on null, since typeof null is "object". See [this interesting article](https://2ality.com/2013/10/typeof-null.html) for detailed explanation)

Reply:
Updated, thank you for the insight